api/types.proto

```diff
@@ -696,3 +696,9 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+message Device {
+	string source = 1;
+	string target = 2;
+	string permissions = 3;
```

Comment:
This function transforms a DeviceMapping to a specs.Device and specs.DeviceCgroup pair:
https://github.com/docker/docker/blob/v1.12.0/daemon/container_operations_unix.go#L324-L371
The interesting part is that it's handling symlinks and folders. With this high-level approach, we can ask to mount `/dev/snd` or `/dev/dri` and all devices will be imported automatically, it will work fine on machines with different hardware.
Whereas if we require to list all the major/minor device numbers in the API, it will be less portable. For instance, for our device `/dev/nvidia-uvm', the major number can be different on between machines.

Also, I feel like I'm missing something since the agent is using the engine-api, if the API was requiring major/minor numbers, we would still need a way to find the device path to create a DeviceMapping object:
https://github.com/flx42/swarmkit/blob/3b4c9af3e6c0953c387db776b9285116231926db/agent/exec/container/container.go#L306-L310

My colleague @3XX0 might want to chime in here.